protocol/v1/qbft/validation/preprepare/validate.go

```diff
@@ -27,6 +34,190 @@ func ValidatePrePrepareMsg(resolver LeaderResolver) pipelines.SignedMessagePipel
 		if uint64(signers[0]) != leader {
 			return fmt.Errorf("proposal leader invalid")
 		}
-		return nil
+
```

Comment:
why moving this validation here?

Reply:
@nivBlox To validate it before adding message to container.